src/CommunityToolkit.Maui/Essentials/FileFolderDialog.android.cs

```diff
@@ -0,0 +1,435 @@
+			}
+
+			Log.Debug("~~~~", "m_dir=" + directory);
+			if (!directoryFile.Exists() || !directoryFile.IsDirectory)
+			{
+				return directories;
+			}
+
+			foreach (var file in directoryFile.ListFiles() ?? Array.Empty<File>())
+			{
+				if (file.IsDirectory)
+				{
+					// Add "/" to directory names to identify them in the list
+					directories.Add(file.Name + "/");
+				}
+				else if (fileSelectionMode is FileSelectionMode.FileOpen
+												or FileSelectionMode.FileOpenRoot
+												or FileSelectionMode.FileSave
+												or FileSelectionMode.FileSaveRoot)
+				{
+					// Add file names to the list if we are doing a file save or file open operation
+					directories.Add(file.Name);
+				}
+			}
+		}
+		catch (Exception)
+		{
+			// ignored
+		}
+
+		directories.Sort();
+		return directories;
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////                                   START DIALOG DEFINITION                                    //////
+	//////////////////////////////////////////////////////////////////////////////////////////////////////////
+	AlertDialog.Builder CreateDirectoryChooserDialog(string title,
+		List<string> listItems,
+		EventHandler<DialogClickEventArgs> onClickListener,
+		string defaultFileName = "default")
+	{
+		var dialogBuilder = new AlertDialog.Builder(context);
+		////////////////////////////////////////////////
+		// Create title text showing file select type // 
+		////////////////////////////////////////////////
+		titleView1 = new TextView(context)
+		{
+			LayoutParameters = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.WrapContent),
+
+			//m_titleView1.setTextAppearance(m_context, android.R.style.TextAppearance_Large);
+			//m_titleView1.setTextColor( m_context.getResources().getColor(android.R.color.black) );
+			Text = fileSelectionMode switch
+			{
+				FileSelectionMode.FileOpen or FileSelectionMode.FileOpenRoot => "Open",
+				FileSelectionMode.FileSave or FileSelectionMode.FileSaveRoot => "Save As",
+				FileSelectionMode.FolderChoose or FileSelectionMode.FolderChooseRoot => "Select Folder",
+				_ => throw new NotSupportedException($"{fileSelectionMode} is not yet supported")
+			},
+
+			//need to make this a variable Save as, Open, Select Directory
+			Gravity = GravityFlags.CenterVertical
+		};
+
+		//_mTitleView1.SetBackgroundColor(Color.DarkGray); // dark gray 	-12303292
+		titleView1.SetTextColor(Color.Black);
+		titleView1.SetTextSize(ComplexUnitType.Dip, 18);
+		titleView1.SetTypeface(null, TypefaceStyle.Bold);
+
+		// Create custom view for AlertDialog title
+		var titleLayout1 = new LinearLayout(context)
+		{
+			Orientation = Orientation.Vertical
+		};
+		titleLayout1.AddView(titleView1);
+
+		if (fileSelectionMode is FileSelectionMode.FileSave)
+		{
+			///////////////////////////////
+			// Create New Folder Button  //
+			///////////////////////////////
```

Comment:
This comment isn't helpful; `new Button` is pretty obvious that we're creating a button
```suggestion
```